load_easybuild_module.sh

```diff
@@ -0,0 +1,108 @@
+# Script to load the environment module for a specific version of EasyBuild.
+# If that module is not available yet, the current latest EasyBuild version of EasyBuild will be installed,
+# and used to install the specific EasyBuild version being specified.
+#
+# This script must be sourced, since it makes changes in the current environment, like loading an EasyBuild module.
+
+set -o pipefail
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 <EasyBuild version>" >&2
+    exit 1
+fi
+
+# don't use $EB_VERSION, since that enables always running 'eb --version'
+EB_VERSION=${1}
+
+# make sure that environment variables that we expect to be set are indeed set
+if [ -z "${TMPDIR}" ]; then 
+    echo "\$TMPDIR is not set" >&2
+    exit 2
+fi
+
+if [ -z "${EB}" ]; then 
+    echo "\$EB is not set" >&2
+    exit 2
+fi
+
+# make sure that utility functions are defined (cfr. scripts/utils.sh script in EESSI/software-layer repo)
+type check_exit_code
+if [ $? -ne 0 ]; then
+    echo "check_exit_code function is not defined" >&2
+    exit 3
+fi
+
+echo ">> Checking for EasyBuild module..."
+
+ml_av_easybuild_out=${TMPDIR}/ml_av_easybuild.out
+module avail 2>&1 | grep -i easybuild/${EB_VERSION} &> ${ml_av_easybuild_out}
+
+if [[ $? -eq 0 ]]; then
+    echo_green ">> Module for EasyBuild v${EB_VERSION} found!"
+else
+    echo_yellow ">> No module yet for EasyBuild v${EB_VERSION}, installing it..."
+
+    EB_TMPDIR=${TMPDIR}/ebtmp
+    echo ">> Temporary installation (in ${EB_TMPDIR})..."
+    pip_install_out=${TMPDIR}/pip_install.out
+    pip3 install --prefix ${EB_TMPDIR} easybuild &> ${pip_install_out}
+
+    # keep track of original $PATH and $PYTHONPATH values, so we can restore them
+    ORIG_PATH=${PATH}
+    ORIG_PYTHONPATH=${PYTHONPATH}
+
+    echo ">> Final installation in ${EASYBUILD_INSTALLPATH}..."
+    export PATH=${EB_TMPDIR}/bin:${PATH}
+    export PYTHONPATH=$(ls -d ${EB_TMPDIR}/lib/python*/site-packages):${PYTHONPATH}
+    eb_install_out=${TMPDIR}/eb_install.out
+    ok_msg="Latest EasyBuild release installed, let's go!"
+    fail_msg="Installing latest EasyBuild release failed, that's not good... (output: ${eb_install_out})"
+    ${EB} --install-latest-eb-release 2>&1 | tee ${eb_install_out}
```

Comment:
Is it necessary to install a module for the latest release?

Maybe we don't want that version (may not hurt, but might be confusing). If the `eb` installed with `pip` is the latest release we have two latest installations, but only use the `pip`-installed `eb` for installing the latest release and the required version.

Reply:
It's not strictly necessary to install the latest release, but keep in mind there's 2 scenarios here:

1) If the requested EasyBuild version is an older version, say `4.5.0`, since installing the current latest version (`4.7.2`) as a module is overkill, since `EasyBuild/4.5.0` can also be installed with the temporary installed that was obtained with `pip`.
2) If the requested EasyBuild version is the latest release, then the temporary EasyBuild installation with `pip` does not include an easyconfig file for that version of EasyBuild (chicken-egg problem: the release has to be available before an easyconfig file can be created for it).

Scenario 2 is why we use `eb --install-latest-eb-release`, since that download the easyconfig file for the latest EasyBuild release from [the `develop` branch of the central easyconfigs repository](https://github.com/easybuilders/easybuild-easyconfigs/tree/develop/easybuild/easyconfigs/e/EasyBuild) (having an easyconfig file available there for the latest EasyBuild release is a part of the procedure for releasing EasyBuild).

So while in scenario 1 the module for the latest EasyBuild release obtained via `eb --install-latest-eb-release` is overkill, it's necessary in scenario 2 (and then the `eb --search` is futile, the installation using an easyconfig file that's included with EasyBuild can't be done).

We can consider changing this approach, but the benefit would be very small, and I think that should be done in a follow-up PR. The whole point of this PR is to put the installation procedure of EasyBuild in a stand-alone script, so it can be used to kickstart a new EESSI pilot version.

Reply:
Thanks a lot for the explanation! Fully agree that we could change the approach in a follow-up PR. So no need to change it here.

If we want to change it, we could do something like this:
1. `pip install` latest `eb` release.
2. check if the required version ($EB_VERSION) can be installed with the `pip`-installed `eb`, e.g., run `eb --search EasyBuild-${EB_VERSION}.eb` or similar. if so, install it.
3. if not, check if the `pip`-installed `eb` has the version that is required, if so run `eb --install-latest-eb-release ...`